### DISCUSSION TITLE
Targets: energy weighting option

### DISCUSSION THREAD

**Discussion by ragg987** [BottlecapDave/HomeAssistant-OctopusEnergy#721] [Ideas]
A slightly edge requirement, but valuable for my use case. Can the target sensor have an option for energy demand weighting in the context of an Octopus Agile tariff?

An example:
I have a heat-pump for hot water and heating. This is one of my big energy consumers, so a focus for optimisation.

For space heating, when I first switch the HP on it ramps up power to heat the large volume of water (typically around 2 to 3kW) and after 30 minutes to 1hr it gets to a stable state and will run at somewhere between 0.7 and 2kW depending on demand.

So in an ideal world the start window would recognise the power demand profile and use a weighting to seek the optimum start time based on cost.

Other use cases would be around washing machine (~70% of energy in first 30 minutes of a 3hr cycle) or dishwasher (~50% of energy in first 30 mins, then ~30% 2 hours later).

I have seen your comments about the target logic getting too complex, but thought I would put this out there anyways.

**Comment by BottlecapDave** [comment]
Thanks for the suggestion. I think this sounds like a good addition. I've added https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/733 to cover this, but I'm not sure when I'll get around to it.
